PythonTool/TagExpansion.py

```diff
@@ -61,70 +58,53 @@ def get_vector_tags(mb, element):
 
     # If there are none of the specified mesh elements, print a warning and exit.
     if len(element_list) == 0:
-        print("WARNING: No hex elements were found in this data file.")
+        print("WARNING: No elements of this type were found in the data file.")
         exit()
 
     tag_list = mb.tag_get_tags_on_entity(element_list[0])
 
-    # Check if each tag is a vector. If so, append it to the list.
+    # Check the type of each tag and append to the appropriate list.
+    scalar_tags = []
     vector_tags = []
     for tag in tag_list:
         tag_length = mb.tag_get_length(tag)
         if tag_length > 1:
             vector_tags.append(tag)
+        else:
+            scalar_tags.append(tag)
 
-    return element_list, vector_tags
+    return element_list, scalar_tags, vector_tags
 
 
-def tag_expansion(mesh_file, output_file = None):
+def tag_expansion(mesh_file, mb_ref, mb_exp, hexes, scal_tags, vec_tag, length, name):
     """
-    Expand the vector tags on each element in the given mesh data file. Write a
-    file to disk for each index with the corresponding scalar tag values.
+    Expand the vector tag on each element in the given mesh data file. Write a
+    file to disk for each index with the corresponding scalar tag value.
 
     Input:
     ______
        mesh_file: str
            User supplied mesh file location.
-       output_file: str
-           Optional user supplied output file name.
+       mb_ref: Core
+           A PyMOAB core instance with a loaded data file for reference.
+       mb_exp: Core
+           A PyMOAB core instance with a loaded data file for expanding vector tags.
+       hexes: List
+           A list of all hex elements in the mesh.
+       scal_tags: List
+           A list of all scalar tags in the mesh.
+       vec_tags: List
+           A list of all vector tags in the mesh.
+       length: int
```

Comment:
I don't think this needs to be passed in since you get the `vec_tags` list as input, you can get the length when it's passed. Should minimize possible errors with passing around information.